src/bindings/js/node/src/infer_request.cpp

```diff
@@ -48,7 +55,7 @@ Napi::Object InferRequestWrap::Wrap(Napi::Env env, ov::InferRequest infer_reques
 }
 
 void InferRequestWrap::set_tensor(const Napi::CallbackInfo& info) {
-    if (info.Length() != 2 || !info[0].IsString() || !info[1].IsObject()) { 
+    if (info.Length() != 2 || !info[0].IsString() || !info[1].IsObject()) {
```

Comment:
Lines which "check key type and value" are recurring many times over and over.

I suggest making helper functions to replace them, reducing complexity and possible errors:
```cpp
// Example:
bool is_single_tensor(const Napi::CallbackInfo& info) {
    return info.Length() == 1 && info[0].IsObject();
}

// And add more like this for int etc.
bool is_key_str_pair(const Napi::CallbackInfo& info) {
    return info.Length() == 2 && info[0].IsString() && info[1].IsObject();
}

// or alternative with templates:

template<typename T>
bool check_key_tensor_pair<T>(const Napi::CallbackInfo& info) {
    // raise error that this overload is not existing? Or even drop this part.
};

template<>
bool check_key_tensor_pair<std::string>(const Napi::CallbackInfo& info) {
    return info.Length() == 2 && info[0].IsString() && info[1].IsObject();
}

template<>
bool check_key_tensor_pair<int>(const Napi::CallbackInfo& info) {  // int or int32_t or size_t, up to design
    return info.Length() == 2 && info[0].IsNumber() && info[1].IsObject();
}

```
I will leave naming, impl details and number of overloads to you.